modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -228,30 +229,24 @@
                  :standard-deviation-aggregations]]
   (defmethod driver/supports? [:mongo feature] [_driver _feature] true))
 
-(defn- db-version [db]
-  (get-in db [:details :version]))
+(defmethod driver/database-supports? [:mongo :expressions]
+  [driver _feature db]
+  (boolean (some-> (driver/dbms-version driver db)
+                   :semantic-version
+                   (driver.u/semantic-version-gte [4]))))
 
-(defn- parse-version [version]
-  (->> (str/split version #"\.")
-       (take 2)
-       (map #(Integer/parseInt %))))
-
-(defn- db-major-version [db]
-  (some-> (db-version db) parse-version first))
-
-(defmethod driver/database-supports? [:mongo :expressions] [_ _ db]
-  (let [version (db-major-version db)]
-    (and (some? version) (>= version 4))))
-
-(defmethod driver/database-supports? [:mongo :date-arithmetics] [_ _ db]
-  (let [version (db-major-version db)]
-    (and (some? version) (>= version 5))))
+(defmethod driver/database-supports? [:mongo :date-arithmetics]
+  [driver _feature db]
+  (boolean (some-> (driver/dbms-version driver db)
```

Comment:
The expectation here is that you're supposed to check `(:dbms_version db)`, which gets populated during the sync process with the results of running the `driver/dbms-version` query, instead of running that query every time we do a feature check. This is the impl of `dbms-version` for Mongo:

```clj
(defmethod driver/dbms-version :mongo
  [_ database]
  (with-mongo-connection [^com.mongodb.DB conn database]
    (let [build-info (mg/command conn {:buildInfo 1})]
      {:version (get build-info "version")
       :semantic-version (get build-info "versionArray")})))
```


It's going to be very slow and expensive to be opening a connection and getting version information every time we want to do a feature support check -- this can happen multiple times in one query.